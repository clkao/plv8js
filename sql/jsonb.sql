CREATE SCHEMA plv8;
CREATE FUNCTION get_key(key text, json_raw jsonb) RETURNS jsonb
LANGUAGE plv8 IMMUTABLE STRICT
AS $$
  var val = json_raw[key];
  var ret = {};
  ret[key] = val;
  return ret;
$$;

CREATE TABLE jsononly (
    data jsonb
);

COPY jsononly (data) FROM stdin;
{"ok": true}
\.

-- Call twice to test the function cache.
SELECT get_key('ok', data) FROM jsononly;
SELECT get_key('ok', data) FROM jsononly;
