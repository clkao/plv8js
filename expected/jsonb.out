CREATE SCHEMA
CREATE FUNCTION
CREATE TABLE
COPY 1
   get_key    
--------------
 {"ok": true}
(1 row)

   get_key    
--------------
 {"ok": true}
(1 row)

